internal/controller/monovertexrollout/monovertexrollout_progressive.go

```diff
@@ -48,64 +49,98 @@ func (r *MonoVertexRolloutReconciler) AssessUpgradingChild(
+			return apiv1.AssessmentResultUnknown, "", nil
 		}
 
-		// if end time has arrived (i.e. we continually determined "Success" for the entire prescribed period of time),
-		// if we need to launch an AnalysisRun, we can do it now;
-		// otherwise, we can declare success
-		if childStatus.BasicAssessmentEndTimeArrived() {
-			analysis := mvtxRollout.GetAnalysis()
-			// only check for and create AnalysisRun if templates are specified
-			if len(analysis.Templates) > 0 {
-				// this will create an AnalysisRun if it doesn't exist yet; or otherwise it will check if it's finished running
-				analysisStatus, err := progressive.PerformAnalysis(ctx, existingUpgradingChildDef, mvtxRollout, mvtxRollout.GetAnalysis(), mvtxRollout.GetAnalysisStatus(), r.client)
-				if err != nil {
-					return apiv1.AssessmentResultUnknown, "", err
-				}
-				return progressive.AssessAnalysisStatus(ctx, existingUpgradingChildDef, analysisStatus)
-			} else {
-				return apiv1.AssessmentResultSuccess, "", nil
+		// if we succeed, we must continue to succeed for a prescribed period of time to consider the resource health
+		// check "successful".
+		if assessment == apiv1.AssessmentResultSuccess {
+			if !childStatus.IsTrialWindowStartTimeSet() {
+				_ = progressive.UpdateUpgradingChildStatus(mvtxRollout, func(status *apiv1.UpgradingChildStatus) {
+					status.TrialWindowStartTime = &metav1.Time{Time: currentTime}
+					status.AssessmentResult = apiv1.AssessmentResultUnknown
+				})
+				numaLogger.Debugf("Assessment succeeded for upgrading child %s, setting TrialWindowStartTime to %s", existingUpgradingChildDef.GetName(), currentTime)
 			}
-		}
 
+			// Check if the trial window is set and if the success window has passed.
+			if childStatus.IsTrialWindowStartTimeSet() && currentTime.Sub(childStatus.TrialWindowStartTime.Time) >= assessmentSchedule.Period {
+				// Success window passed, launch AnalysisRun or declared success
+				_ = progressive.UpdateUpgradingChildStatus(mvtxRollout, func(status *apiv1.UpgradingChildStatus) {
+					status.BasicAssessmentEndTime = &metav1.Time{Time: currentTime}
+					status.BasicAssessmentResult = apiv1.AssessmentResultSuccess
+				})
+				return r.checkAnalysisTemplates(ctx, mvtxRollout, existingUpgradingChildDef)
+			}
+
+			numaLogger.Debugf("Assessment succeeded for upgrading child %s, but success window has not passed yet", existingUpgradingChildDef.GetName())
+			// Still waiting for a success window to pass
+			return apiv1.AssessmentResultUnknown, "", nil
+		}
+	} else {
+		if childStatus.BasicAssessmentResult == apiv1.AssessmentResultSuccess {
+			return r.checkAnalysisTemplates(ctx, mvtxRollout, existingUpgradingChildDef)
+		}
+		return childStatus.BasicAssessmentResult, "Basic assessment failed", nil
 	}
 
 	return apiv1.AssessmentResultUnknown, "", nil
+}
 
+// checkAnalysisTemplates checks if there are any analysis templates to run and runs them if so.
```

Comment:
```suggestion
// checkAnalysisTemplates checks if there are any analysis templates to run and runs them if so.
// otherwise it returns success.
```